requests_cache/serializers/__init__.py

```diff
@@ -1,47 +1,74 @@
-# flake8: noqa: F401
-from logging import getLogger
-from typing import Type, Union
+import pickle
 
 from .. import get_placeholder_class
-from .base import BaseSerializer
-from .pickle_serializer import PickleSerializer, SafePickleSerializer
+from . import preconf
+from .pipeline import SerializerPipeline, Stage
 
-SerializerSpecifier = Union[str, BaseSerializer, Type[BaseSerializer]]
-logger = getLogger(__name__)
+pickle_serializer = pickle
 
 try:
-    from .bson_serializer import BSONSerializer
+    from itsdangerous import Signer
+
+    def safe_pickle_serializer(secret_key=None, salt="requests-cache", **kwargs):
+        if not secret_key:
+            raise ValueError("Cannot use itsdangerous without a secret key!")
+        return SerializerPipeline(
+            [
+                pickle_serializer,
+                Stage(Signer(secret_key=secret_key, salt=salt), dumps='sign', loads='unsign'),
+            ],
+        )
+
+
 except ImportError as e:
-    BSONSerializer = get_placeholder_class(e)  # type: ignore
+    safe_pickle_serializer = get_placeholder_class(e)
+
+
 try:
-    from .json_serializer import JSONSerializer
-except ImportError as e:
-    JSONSerializer = get_placeholder_class(e)  # type: ignore
+    import ujson as json
 
+    json_serializer = SerializerPipeline(
+        [
+            preconf.ujson_converter,  # CachedResponse -> JSON
+            json,  # JSON -> str
+        ],
+    )
+except ImportError:
+    import json
 
-SERIALIZER_CLASSES = {
-    'bson': BSONSerializer,
-    'json': JSONSerializer,
-    'pickle': PickleSerializer,
-    'safe_pickle': SafePickleSerializer,
-}
+    json_serializer = SerializerPipeline(
+        [
+            preconf.json_converter,  # CachedResponse -> JSON
+            json,  # JSON -> str
+        ],
+    )
 
+try:
+    import bson.json_util
+
+    bson_serializer = SerializerPipeline(
+        [
+            preconf.bson_converter,  # CachedResponse -> BSON
+            bson.json_util,  # BSON -> str
+        ],
+    )
+except ImportError as e:
+    bson_serializer = get_placeholder_class(e)
 
-def init_serializer(serializer: SerializerSpecifier = None, *args, **kwargs) -> BaseSerializer:
-    """Initialize a serializer from a name, class, or instance"""
-    logger.debug(f'Initializing serializer: {serializer}')
+try:
+    import pyyaml
```

Comment:
The package on PyPI is `pyyaml`, but the module name is just `yaml`.
Also, it would be worth adding this to the optional dependencies:
```
poetry add --optional pyyaml
```